model/canvas.cc

```diff
@@ -168,6 +168,10 @@ namespace minsky
         if (item)
           {
             item->resize(lasso);
+            // resize both operator and variable in coupled integral variable. for feature 94
+            if (auto i=dynamic_cast<IntOp*>(item.get()))
+               if (i->coupled())
+                 i->intVar->resize(lasso);
```

Comment:
Would this work correctly? ISTM, the lasso should refer to resizing the integral op, and then the intVar needs to be scaled accordingly.

Reply:
I have found no other way to scale the variable otherwise. Issuing cairo_scale() in RenderVariable::draw() was a bust, adding cairo_scale() in operation.cc where there are if (i->coupled()) has also been fruitless. I also added intop casts in VariableBase::draw() to scale the attached variable which is accessed via i->intVar, and this also had no effect. This latest attempt seems to work, but there seems to be something wrong with the outport and the size of the bounding box, which is what you may be referring to.

Reply:
Why wouldn't you do something like:
```
float oldIWidth=iWidth();
iWidth(std::abs(b.x1-b.x0)*invZ);
intVar->iWidth(intVar->iWidth() * iWidth/oldOWidth);
```
ditto for iHeight

Reply:
OK. Not sure if you meant inside canvas.cc? I did this:

      case LassoMode::itemResize:
        if (item)
          {
            item->resize(lasso);
            // resize both operator and variable in coupled integral variable. for feature 94
            if (auto i=dynamic_cast<IntOp*>(item.get()))
               if (i->coupled()) {
				  float invZ=1.0/i->zoomFactor(); 
                  float oldIWidth=i->iWidth();
                  i->iWidth(std::abs(lasso.x1-lasso.x0)*invZ);
                  i->intVar->iWidth(i->intVar->iWidth() * i->iWidth()/oldIWidth);
                  
                  float oldIHeight=i->iHeight();
                  i->iHeight(std::abs(lasso.y1-lasso.y0)*invZ);
                  i->intVar->iHeight(i->intVar->iHeight() * i->iHeight()/oldIHeight);
			  }
               
            //     i->intVar->resize(lasso);
            requestRedraw();
          }
        break;

But this again only resizes the integral... Perhaps you meant inside VariableBase::resize()?

Reply:
Great, thanks. This seems to work:

              float oldIWidth=i->iWidth();
              i->intVar->iWidth(std::abs(lasso.x1-lasso.x0)*invZ);
              i->intVar->iWidth(i->intVar->iWidth() * i->iWidth()/oldIWidth);

and likewise for the height.

Just fixing the scaling of operator symbols within boxes now.